src/Button/Button.tsx

```diff
@@ -0,0 +1,75 @@
+import { ButtonHTMLAttributes, FC, forwardRef, ReactNode } from 'react';
+import { BackgroundColorProps } from 'styled-system';
+
+import ButtonContent from './ButtonContent';
+import ButtonSpinner from './ButtonSpinner';
+import { Size } from './sizes';
+import StyledButton from './StyledButton';
+import Text from '../Text';
+import { Variant } from './variants';
+import { Intent } from '../theme/colors/intent';
+import { PaletteColor } from '../theme/colors/palette';
+
+type Props = ButtonHTMLAttributes<HTMLButtonElement> &
+  BackgroundColorProps & {
+    fullWidth?: boolean | boolean[];
+    iconStart: ReactNode;
+    iconEnd: ReactNode;
+    intent?: Intent;
+    paletteColor?: PaletteColor;
+    size?: Size | Size[];
+    spin?: boolean;
+    variant?: Variant;
+  };
+
+const Button: FC<Props> = forwardRef<HTMLButtonElement, Props>(
+  (
+    {
+      bg = null,
+      children,
+      disabled = false,
+      fullWidth = false,
+      iconEnd,
+      iconStart,
+      intent = 'brand',
+      paletteColor = '',
+      size = 'large',
+      spin = false,
+      type,
+      variant = 'primary',
+      ...props
+    },
```

Comment:
I think I've faithfully reproduced the prop types and defaults here, but this is a particularly good case for an extra set of eyes.  Note that I had to explicitly call out a few more props than the JS version in order to default them here.  They've been passed through explicitly as well.

Reply:
other than my comment above re: `iconStart` and `iconEnd` not being optional anymore, the rest looks right to me